### PR TITLE
Create user model, set up login and register route

### PR DESCRIPTION
**Pull request by oliviajohansen**
- Use `auth` in routes that allow the creation, updating and deletion of models
- Implement `User` model that stores email and password for users
- Implement `/register` and `/login` route for creation of users and tokens
- Add `userId` to `request` after authentication so that it can be used later on if needed

### CODE REVIEW

File: src/models/User.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+class User extends Model {
+  public id!: number;
+  public email!: string;
+  public password!: string;
+}
+
+User.init(
+  {
+    userId: {
```
Comment (ivantjh) [inline src/models/User.ts]: Should this be `id`?

File: src/models/User.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      type: DataTypes.INTEGER,
```
Comment (ivantjh) [inline src/models/User.ts:25]: Make the `init` function and the migration portion the same

File: src/middleware/auth.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const jwt = require('jsonwebtoken');
+import User from '../models/User';
+import {Request, Response, NextFunction} from 'express';
+
+module.exports = function (req: Request, res: Response, next: NextFunction) {
```
Comment (ivantjh) [inline src/middleware/auth.ts]: Use typescript syntax, `export default ...`

**Review by ivantjh — CHANGES_REQUESTED** [review]

File: src/models/User.ts (hunk 1 above)
Comment (oliviajohansen) [reply src/models/User.ts]: ok

File: src/models/User.ts (hunk 2 above)
Comment (oliviajohansen) [reply src/models/User.ts:25]: are you referring to the diff names of userId: userId in user model and id in migration file and making them the same?

File: src/middleware/auth.ts (hunk 3 above)
Comment (oliviajohansen) [reply src/middleware/auth.ts]: ok thanks

File: src/models/User.ts (hunk 2 above)
Comment (zatkiller) [reply src/models/User.ts:25]: Yea ps this one should be id too, my bad

**Comment by ivantjh** [comment]
@oliviajohansen, @zatkiller, help me check again! Not sure if I missed out anything

**Review by zatkiller — APPROVED** [review]
LGTM

**Pull request merged**